src/main/java/bio/terra/service/snapshot/flight/delete/SnapshotDeleteFlight.java

```diff
@@ -143,6 +134,17 @@ public SnapshotDeleteFlight(FlightMap inputParameters, Object applicationContext
                 azureAuthService)),
         randomBackoffRetry);
 
+    // Delete access control first so Readers and Discoverers can no longer see snapshot
+    // Google auto-magically removes the ACLs from BQ objects when SAM
+    // deletes the snapshot group, so no ACL cleanup is needed beyond that.
+    addStep(new DeleteSnapshotAuthzResource(iamClient, snapshotId, userReq));
+
+    // Now that we no longer have the resources in SAM, we can delete the underlying Sam group
+    // that was created for snapshots byRequestId
+    // Note: Must delete primary data before metadata; it relies on being able to retrieve the
+    // snapshot object from the metadata to know what to delete.
+    addStep(new DeleteSnapshotDeleteSamGroupStep(iamClient, snapshotRequestDao, snapshotId));
+
```

Comment:
Goal of this is to move the Sam Resource delete towards the end of the delete flight. There is a series of steps that rely on each other that prevent us from moving this any further towards the end without reworking the steps. 

I think moving these steps to after the primary data deletions steps will still be helpful because it's likely the primary data deletion steps are relatively more flaky.